src/main/java/org/folio/list/services/EntityManagerFlushService.java

```diff
@@ -0,0 +1,23 @@
+package org.folio.list.services;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class EntityManagerFlushService {
+
+  @PersistenceContext
+  private final EntityManager entityManager;
+
+  public void flush() {
+    synchronized (entityManager) {
+      entityManager.flush();
+      entityManager.clear();
+    }
```

Comment:
This is synchronized so that we don't need to worry about any new actions getting added to the entity manager between the `flush()` and `clear()` calls